core/app/models/spree/stock/coordinator.rb

```diff
@@ -66,12 +65,19 @@ def build_packages(packages = Array.new)
 
       private
 
-      def unallocated_inventory_units
-        inventory_units - @preallocated_inventory_units
+      def stock_location_variant_ids
+        location_variant_ids = StockItem.where(variant_id: unallocated_variant_ids).joins(:stock_location).merge(StockLocation.active).pluck(:stock_location_id, :variant_id)
```

Comment:
I feel like this should be moved off to a scope and then to make things more readable:

```ruby
stock_location_ids, variant_ids = StockItem.stock_location_and_variant_ids
location_lookup = StockLocation.where(id: stock_location_ids.uniq).map { |l| [l.id, l] }.to_h